### PR TITLE
SynStream doesn't set an associated_to_stream_id.

### PR DESCRIPTION
**Pull request by authorNari**

### CODE REVIEW

File: lib/spdy/protocol.rb [hunk 1]
```diff
@@ -83,6 +83,9 @@ class SynStream < BinData::Record
 
         def create(opts = {})
           build({:type => 1, :len => 10}.merge(opts))
+          if opts[:associated_to_stream_id]
```
Comment (igrigorik) [inline lib/spdy/protocol.rb]: This should be moved into the build method - we already initialize and check specific variables there. Otherwise, lgtm!

**Comment by authorNari** [comment]
OK! :smile: 
Is ce31d9d acceptable?

**Comment by igrigorik** [comment]
Errr, sorry @authorNari, now I feel silly.. didn't read the code correctly on this end. On second pass, looks like your first approach is actually what we wanted, albeit with a small modification. How about the following:

```
def create(opts = {})
  build({:type => 1, :len => 10}.merge(opts))
  self.associated_to_stream_id = opts.fetch(:associated_to_stream_id, nil)
end
```

Would that work? Apologies about the mixup.

**Comment by authorNari** [comment]
No problem! :octocat: 
Is 2c7d04c acceptable?
I think we can't set nil on assoc_to_stream_id.

**Comment by igrigorik** [comment]
lgtm! Thanks for your patience. :-)

**Pull request merged**